dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/DefaultTrackerObjectsDeletionService.java

```diff
@@ -75,7 +75,10 @@ public TrackerTypeReport deleteEnrollments(TrackerBundle bundle) {
 
       Entity objectReport = new Entity(TrackerType.ENROLLMENT, uid);
 
-      Enrollment enrollment = enrollmentService.getEnrollment(uid);
+      Enrollment enrollment = manager.get(Enrollment.class, uid);
```

Comment:
I know I did not for events but I am wondering if we should use the exporter service here @enricocolasante ?

Reply:
I think we should use the exporter here, the thing is using the service-tracker `enrollmentService`, we run the validations and map the Enrollment to a new object, which is then used by this class to delete the enrollment, and fails because the session already contains a different Enrollment object with the same id.
Long story short, we need to refactor this logic and it might need its own PR.

Reply:
thank you for checking 😄

Reply:
I think we need to discuss on how to proceed here.
I am not sure we should use the exporter service here. `deleteEnrollments` method should never be exposed and it should only be used by our importer that already did all the needed validations to check if the enrollment is there and if it can be deleted by the user (and also if the events can be deleted).

Here in the end we should just update the enrollment to `deleted=false` and all the lastUpdated info.